src/games/domino/training/alpha_zero_trainer.py

```diff
@@ -0,0 +1,363 @@
+            If there is not enough saved data, missing data will be generated via self play.
+        param sample: int:
+            Size of NN input in training
+        param tag: str
+            Tag of the checkpoint to be saved
+        param num_process: int
+            Number of parallel self_play games. Set to one to disable parallelism.
+        param verbose: bool
+            Set True to enable verbose log
+        param save_data: bool
+            Set True to save generated training data
+
+        return
+            (total_loss, loss_policy, loss_value)
+        """
+        if verbose:
+            print('Getting training data...')
+        data = self._get_data(simulate, num_process, verbose, self.batch_size)
+        
+        if save_data:
+            num_files = len([name for name in os.listdir(self.data_path) if os.path.isfile(f'{self.data_path}/{name}')])
+            path = f'{self.data_path}/training_data_{num_files}.json'
+            with open(path, 'w') as writer:
+                json.dump(data, writer)
+            if verbose:
+                print(f'Training data saved at {path}')
+
+        if verbose:
+            print(f'[Epoch {epoch}] -- Training net --')
+            start = time.time()
+
+        self.adjust_learning_rate(epoch, self.net.optimizer)
+        total_loss, policy_loss, value_loss = 0,0,0
+        batch_size = len(data)
+        total = 0
+
+        for _ in range(batch_size // sample):
+            batch = random.sample(data, sample)
+            total += sample
+            loss = self.net.train_batch(batch)
+            total_loss += loss[0]
+            policy_loss += loss[1]
+            value_loss += loss[2]
+
+        loss = (total_loss / total, policy_loss / total, value_loss / total)
+        self.error_log.append(loss)
+
+        config = self.build_config(sample, tag, epoch)
+        if self.loss > loss[0]:
+            self.loss = loss[0]
+            self.net.save(self.error_log, config, epoch, self.save_path, True, tag + '-min', verbose=True)
+        else:
+            self.net.save(self.error_log, config, epoch, self.save_path, False, tag, verbose=True)
+
+        if verbose:
+            print(f'-- Training took {str(int(time.time() - start))} seconds --')
+            print(f'policy head loss: {loss[1]} -- value head loss: {loss[2]} -- TOTAL LOSS: {loss[0]}')
+            print('')
+
+        return loss
+
+    def _get_data(
+        self,
+        simulate: bool,
+        num_process: int,
+        verbose: bool,
+        batch_size: int
+    ):
+        data = []
+        num_games = 0
+
+        if simulate:
+            if verbose:
+                print('Simulations started')
+                start = time.time()
+
+            if num_process > 1:
+                while len(data) < batch_size:
+                    jobs = [(self.handouts, self.rollouts, self.alpha)] * (batch_size // num_process)
+                    num_games += batch_size / num_process
```

Comment:
```suggestion
                    num_games += batch_size // num_process
```
Or save the value in a variable before the `while`